Fixing truncate problem

architeacher opened the pull request:
When passing a length that is equal to the length of ellipsis, the returned string will be the ellipsis value only, this would happen either passing TRUE of FALSE for the exact option.
e.g.

String::truncate('Hello', 3);  // would return ...

architeacher committed:
Fixing truncate problem

When passing a length that is equal to the length of ellipsis, the returned string will be the ellipsis value only, this would happen either passing TRUE of FALSE for the exact option.
e.g.

String::truncate('Hello', 3);  // would return ...

markstory commented:
Your change is causing tests to fail. Existing tests will need to be updated.

markstory commented on src/Utility/Text.php:
```diff
@@ -590,6 +590,11 @@ public static function truncate($text, $length = 100, array $options = [])
             $truncate = mb_substr($truncate, 0, $spacepos);
         }
+        // If truncate still empty, then we don't need to count ellipsis in the cut.
+        if(empty($truncate)){
```
What if truncate is '0'?

ahmedkamals replied:
You are right, I did not think about this scenario, I will modify unit tests as well.

architeacher committed:
Saving up

Fixing the issue when the $truncate is '0' at method Text::truncate around line 594.

architeacher committed:
Fixing truncate problem

dereuromark commented on src/Utility/Text.php:
```diff
@@ -590,6 +590,11 @@ public static function truncate($text, $length = 100, array $options = [])
             $truncate = mb_substr($truncate, 0, $spacepos);
         }
+        // If truncate still empty, then we don't need to count ellipsis in the cut.
+        if(mb_strlen($truncate) == 0){
```
Please respect CS, also it should be === 0

ahmedkamals replied:
Thanks alot, is CS means Coding Standards?
What is the Coding Standard that CakePHP uses?

architeacher committed:
Fixing truncate problem

dereuromark commented:
The report is in the travis build (click on the red X): https://travis-ci.org/cakephp/cakephp/jobs/58429117

See http://book.cakephp.org/3.0/en/contributing/cakephp-coding-conventions.html for CS

architeacher committed:
Applying coding standards.

architeacher committed:
Fixing unit test failure.

markstory commented:
Tests fixed and changed merged. Thanks :+1:
